stance/skeleton/skeleton.py

```diff
@@ -29,13 +30,18 @@ def cos_similarity(self, other: 'SKVector') -> float:
             cosine similarity between self and other. Approaches -1 or 1 for linearly
             dependent vectors and 0 for completely orthogonal vectors
         """
-        magnitude = np.linalg.norm(self.vec) * np.linalg.norm(other.vec)
+        magnitude = self.magnitude * np.linalg.norm(other.vec)
         return np.dot(self.vec, other.vec) / magnitude
 
-
     def __repr__(self) -> str:
         return 'SKVector: {} -> {}'.format(tuple(self.head), tuple(self.tail))
 
+    def _magnitude(self) -> float:
```

Comment:
These are one line methods, you could probably just put this into the constructor, i.e 
`self.magnitude = np.linalg.norm(self.vec)`
up to you though